File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType16RecordCreateForm.cs
using System;
using Neurotec.Biometrics.Standards;
using Neurotec.IO;
using Neurotec.Images;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType16RecordCreateForm : ANRecordCreateForm
	{
		#region Public constructor

		public ANType16RecordCreateForm()
		{
			InitializeComponent();

			tbUdi.MaxLength = ANType16Record.MaxUserDefinedImageLength;
		}

		#endregion

		#region Private properties

		private string Udi
		{
			get
			{
				return tbUdi.Text;
			}
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			ANType16Record record;
			if (imageLoader.CreateFromImage)
			{
				record = new ANType16Record(ANTemplate.VersionCurrent, Idc, Udi, imageLoader.Src,
					imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, imageLoader.Image);
			}
			else if (imageLoader.CreateFromData)
			{
				using (NImage image = NImage.FromMemory(imageLoader.ImageData))
				{
					record = new ANType16Record(ANTemplate.VersionCurrent, Idc, Udi, imageLoader.Src,
						imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, image)
					{
						HorzLineLength = imageLoader.Hll,
						VertLineLength = imageLoader.Vll,
						HorzPixelScale = imageLoader.Hps,
						VertPixelScale = imageLoader.Vps,
						BitsPerPixel = imageLoader.Bpx,
						ColorSpace = imageLoader.Colorspace
					};
				}
			}
			else
			{
				throw new NotSupportedException();
			}

			template.Records.Add(record);
			return record;
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType16RecordCreateForm.vb
Imports System
Imports Neurotec.Biometrics.Standards
Imports Neurotec.IO
Imports Neurotec.Images

Namespace RecordCreateForms

	Partial Public Class ANType16RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public constructor"

		Public Sub New()
			InitializeComponent()

			tbUdi.MaxLength = ANType16Record.MaxUserDefinedImageLength
		End Sub

#End Region

#Region "Private properties"

		Private ReadOnly Property Udi() As String
			Get
				Return tbUdi.Text
			End Get
		End Property

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim record As ANType16Record
			If imageLoader.CreateFromImage Then
				record = New ANType16Record(ANTemplate.VersionCurrent, Idc, Udi, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, imageLoader.Image)
			ElseIf imageLoader.CreateFromData Then
				Using image As NImage = NImage.FromMemory(imageLoader.ImageData)
					record = New ANType16Record(ANTemplate.VersionCurrent, Idc, Udi, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, image) With {.HorzLineLength = imageLoader.Hll, .VertLineLength = imageLoader.Vll, .HorzPixelScale = imageLoader.Hps, .VertPixelScale = imageLoader.Vps, .BitsPerPixel = imageLoader.Bpx, .ColorSpace = imageLoader.Colorspace}
				End Using
			Else
				Throw New NotSupportedException()
			End If

			template.Records.Add(record)
			Return record
		End Function

#End Region
	End Class
End Namespace
